qcodes/dataset/measurements.py

```diff
@@ -721,6 +732,27 @@ def _paramspecbase_from_strings(
 
         return tuple(depends_on), tuple(inf_from)
 
+    def register_parent(
+            self: T, parent: DataSet, link_type: str,
```

Comment:
So currently `link_type` can be anything?

Reply:
Yes. It's completely the responsibility of the users to ensure that they always write `"analysis"` and never `"analsyis"`, or at least that they can handle that ambiguity. Do you think we can do better than this?

Reply:
I think its a good idea to normalize but on the other hand i have no idea of the relevant types so lets leave it for now